src/_pytest/config/__init__.py

```diff
@@ -1078,12 +1085,33 @@ def _checkversion(self):
                     )
                 )
 
-    def _validatekeys(self):
+    def _validate_keys(self) -> None:
         for key in sorted(self._get_unknown_ini_keys()):
-            message = "Unknown config ini key: {}\n".format(key)
-            if self.known_args_namespace.strict_config:
-                fail(message, pytrace=False)
-            sys.stderr.write("WARNING: {}".format(message))
+            self._emit_warning_or_fail("Unknown config ini key: {}\n".format(key))
+
+    def _validate_plugins(self) -> None:
+        required_plugins = sorted(self.getini("required_plugins"))
+        if not required_plugins:
+            return
+
+        plugin_info = self.pluginmanager.list_plugin_distinfo()
+        plugin_dist_names = [dist.project_name for _, dist in plugin_info]
+
+        missing_plugins = []
+        for plugin in required_plugins:
+            if plugin not in plugin_dist_names:
+                missing_plugins.append(plugin)
+
+        if missing_plugins:
+            fail(
```

Comment:
Is it possible the intention was to use `self._emit_warning_or_fail` here?

Reply:
No, we always want to fail ( see earlier in the PR comments for justification )

Reply:
So I was thrown off by the fact that `_emit_warning_or_fail` is extracted to a separate function, and that the help text for `strict_config` was changed. So I'd roll back these two changes -- maybe they were relevant for a previous PR iteration.

Reply:
Why roll back the extraction? I see no harm in leaving it in. Functionality is neither added or removed with it

Reply:
I think because it examines `strict_config` the name is a bad confusing, would call it `_emit_warning_or_fail_if_strict`. But that's a handful, and since it's only used once, I'd remove the indirection and bring it back inline.

Reply:
Perhaps using `warn` as a verb directly, as in `_warn_or_fail_if_strict`, is better?

Reply:
Sure, I’ll change it to the name suggested by @nicoddemus